src/transaction/sign.rs

```diff
@@ -0,0 +1,51 @@
+use async_trait::async_trait;
+use clap::{ArgMatches, Command};
+
+use casper_client::cli::CliError;
+
+use super::creation_common;
+use crate::{command::ClientCommand, common, Success};
+
+pub struct SignTransaction;
+
+#[async_trait]
+impl ClientCommand for SignTransaction {
+    const NAME: &'static str = "sign-transaction";
+    const ABOUT: &'static str =
+        "Read a previously-saved transaction from a file, cryptographically sign it, and output it to a \
+        file or stdout";
+
+    fn build(display_order: usize) -> Command {
+        Command::new(Self::NAME)
+            .about(Self::ABOUT)
+            .display_order(display_order)
+            .arg(
+                common::secret_key::arg(creation_common::DisplayOrder::SecretKey as usize, "")
+                    .required(true),
+            )
+            .arg(creation_common::transaction_path::arg())
+            .arg(creation_common::output::arg())
+            .arg(common::force::arg(
+                creation_common::DisplayOrder::Force as usize,
+                true,
+            ))
+    }
+
+    async fn run(matches: &ArgMatches) -> Result<Success, CliError> {
+        let input_path = creation_common::transaction_path::get(matches);
+        let secret_key = common::secret_key::get(matches).unwrap_or_default();
+        let maybe_output_path = creation_common::output::get(matches).unwrap_or_default();
+        let force = common::force::get(matches);
+        casper_client::cli::sign_transaction_file(input_path, secret_key, maybe_output_path, force)
+            .map(|_| {
+                Success::Output(if maybe_output_path.is_empty() {
```

Comment:
Since this is already in a block expression, could we move the string calculation before the Success::Output? Like
`let output = if maybe_output_path.is_empty() {
String::new()
                } else {
                    format!(
                        "Signed the transaction at {} and wrote to {}",
                        input_path, maybe_output_path
                    )
                }
Success::Output(output)
`

Reply:
I think that this could make the function more readable in general. 
An alternative form that I ended up with looked like this to keep the functions return signature the same:

```
        let output = if maybe_output_path.is_empty(){
            String::new()
        } else {
            format!(
                "Signed the transaction at {} and wrote to {}",
                input_path, maybe_output_path
            )
        };
        casper_client::cli::sign_transaction_file(input_path, secret_key, maybe_output_path, force)
            .map(|_| {
                Success::Output(output)
            })
```

This way we keep the `Result<Success, CliError>`  structure on the last line via map.
I'm okay with either approach.  @gRoussac care to weigh in?

Reply:
Hi, well I am not disturbed by the first version, except indentation, just a small new line make it a bit clearer imo. Otherwise I am am also of course fine to prepare an output var.

```
.map(|_| {
    Success::Output(
        if maybe_output_path.is_empty() {
            String::new()
        } else {
            format!(
                "Signed the transaction at {} and wrote to {}",
                input_path, maybe_output_path
            )
        }
    )
})
```
```
let output = if maybe_output_path.is_empty() {
    String::new()
} else {
    format!(
        "Signed the transaction at {} and wrote to {}",
        input_path, maybe_output_path
    )
};

casper_client::cli::sign_transaction_file(input_path, secret_key, maybe_output_path, force)
    .map(|_| Success::Output(output))
    
```